frontend/appflowy_flutter/lib/mobile/presentation/search/mobile_search_textfield.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy/mobile/presentation/mobile_bottom_navigation_bar.dart';
+import 'package:appflowy_ui/appflowy_ui.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+
+class MobileSearchTextfield extends StatefulWidget {
+  const MobileSearchTextfield({
+    super.key,
+    this.onChanged,
+    required this.hintText,
+    required this.query,
+  });
+
+  final String hintText;
+  final String query;
+  final ValueChanged<String>? onChanged;
+
+  @override
+  State<MobileSearchTextfield> createState() => _MobileSearchTextfieldState();
+}
+
+class _MobileSearchTextfieldState extends State<MobileSearchTextfield> {
+  late final TextEditingController controller;
+  late final FocusNode focusNode;
+  final ValueNotifier<bool> hasFocusValueNotifier = ValueNotifier(true);
+
+  @override
+  void initState() {
+    super.initState();
+    controller = TextEditingController(text: widget.query);
+    focusNode = FocusNode();
+    focusNode.addListener(() {
+      if (!mounted) return;
+      hasFocusValueNotifier.value = focusNode.hasFocus;
+    });
+    controller.addListener(() {
+      if (!mounted) return;
+      widget.onChanged?.call(controller.text);
+    });
+    bottomNavigationBarItemType.addListener(onBackOrLeave);
+    focusNode.requestFocus();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    focusNode.dispose();
+    hasFocusValueNotifier.dispose();
+    bottomNavigationBarItemType.removeListener(onBackOrLeave);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = AppFlowyTheme.of(context);
+    return SizedBox(
+      height: 40,
+      child: ValueListenableBuilder(
+        valueListenable: controller,
+        builder: (context, _, __) {
+          final hasText = controller.text.isNotEmpty;
+          return Row(
+            children: [
+              Expanded(
+                child: TextFormField(
```

Comment:
We have the AFTextField component, but I'm not sure if the designer used it in the figma design.

Reply:
No, they didn't use AFTextField component here in the figma design